server/test/unit/server/db/test_model.py

```diff
@@ -1,19 +1,232 @@
 """
 Tests for the pulp.server.db.model module.
 """
+from tempfile import mkdtemp
+import shutil
+import os
+
+from mock import patch
+
 try:
     import unittest2 as unittest
 except ImportError:
     import unittest
 
-
 from mongoengine import DateTimeField, DictField, Document, IntField, StringField
 
+from pulp.common import error_codes
+from pulp.devel.unit.util import touch
+from pulp.server.exceptions import PulpCodedException
 from pulp.server.db import model
 from pulp.server.db.model.base import CriteriaQuerySet
 from pulp.server.db.model.fields import ISO8601StringField
 
 
+class TestContentUnit(unittest.TestCase):
+    """
+    Test ContentUnit model
+    """
+
+    def setUp(self):
+        self.working_dir = mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.working_dir, ignore_errors=True)
+
+    def test_model_fields(self):
+        self.assertTrue(isinstance(model.ContentUnit.id, StringField))
+        self.assertTrue(model.ContentUnit.id.primary_key)
+
+        self.assertTrue(isinstance(model.ContentUnit.last_updated, IntField))
+        self.assertTrue(model.ContentUnit.last_updated.required)
+
+        self.assertTrue(isinstance(model.ContentUnit.user_metadata, DictField))
+
+        self.assertTrue(isinstance(model.ContentUnit.storage_path, StringField))
+
+        self.assertTrue(isinstance(model.ContentUnit._ns, StringField))
+        self.assertTrue(model.ContentUnit._ns)
+        self.assertTrue(isinstance(model.ContentUnit.unit_type_id, StringField))
+        self.assertTrue(model.ContentUnit.unit_type_id)
+
+    def test_meta_abstract(self):
+        self.assertEquals(model.ContentUnit._meta['abstract'], True)
+
+    def test_post_init_signal_with_unit_key_fields_defined(self):
+        """
+        Test the init signal handler that validates the existence of the
+        unit_key_fields list
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            unit_key_fields = ['id']
+
+        model.ContentUnit.post_init_signal({}, ContentUnitHelper())
```

Comment:
I don't see any assertions in this. I assume the test is so that if the validation fails an exception will be raised and the test will Error. In that case it would be good to catch the expected exception type and then call `self.fail()`. Something should be done to either add an assertion or convert an unexpected error into a graceful failure.